lib/read_parsers.hh

```diff
@@ -22,12 +22,31 @@ namespace read_parsers
 {
 
 struct NoMoreReadsAvailable : public  khmer_exception {
+    explicit NoMoreReadsAvailable(const char *msg) :
+        khmer_exception(msg) {}
+    explicit NoMoreReadsAvailable() :
```

Comment:
"the use of explicit with a default constructor has no effect unless it has default arguments that would allow it to be used in copy-initialization context" http://en.cppreference.com/w/cpp/language/explicit